app/controllers/api/v1/accounts/credentials_controller.rb

```diff
@@ -26,6 +26,7 @@ def account_params
 
   def user_settings_params
     return nil unless params.key?(:source)
```

Comment:
This line can be removed because of the new one

Reply:
params[:source] will be nil if the parameter is not supplied by the requester, checking it #empty? on it will result in 
nil.empty? which will cause it to error.

Reply:
`nil.empty?` is `true`

Reply:
```ruby
return nil if params[:source].nil? || params[:source].empty?
```

Reply:
Another option is to use `blank?` instead of `empty?` because it works on nil and hash, then only one check is needed.